ModularTegustation/lc13_spawners.dm

```diff
@@ -106,75 +106,129 @@
 		/obj/item/tresmetal/indigo = 1
 		)
 
+///This spawner scatters the spawned stuff around where it is placed.
+/obj/effect/spawner/mobspawner
+	icon = 'icons/effects/landmarks_static.dmi'
+	icon_state = "xeno_spawn"
+	///determines how many things to scatter
+	var/max_spawns = 3
+	///determines how big of a range we should scatter things in.
+	var/radius = 1
+	///This weighted list acts as the loot table for the spawner
+	var/list/loot_table
+
+
+/obj/effect/spawner/mobspawner/Initialize()
+	..()
+	if(!length(loot_table))
+		return INITIALIZE_HINT_QDEL
+
+	var/list/candidate_locations = ReturnPlacementTurfs()
+
+	if(!candidate_locations.len)
+		return INITIALIZE_HINT_QDEL
+
+	PlaceMobs(candidate_locations)
+
+	return INITIALIZE_HINT_QDEL
+
+/obj/effect/spawner/mobspawner/proc/ReturnPlacementTurfs()
```

Comment:
This proc notably does not include checks for dense objects.

Reply:
`			continue
		var/bad_turf = FALSE
		for(var/obj/O in turf_in_view)
			if(O.density)
				bad_turf = TRUE
				break
		if(!bad_turf)
			. += turf_in_view`
			Does this work as a solution or should i keep looking into it?